assets/src/scripts/editoria11y-tinymce.js

```diff
@@ -0,0 +1,298 @@
+					return done();
+				}
+				if ( window.Ed11y && ! iframeWin.Ed11y ) {
+					iframeWin.Ed11y = window.Ed11y;
+					return done();
+				}
+				if ( loops2++ > 25 ) {
+					return done();
+				}
+				setTimeout( confirmCtor, 120 );
+			} )();
+		};
+		/**
+		 *
+		 */
+		s.onerror = function () {
+			done();
+		};
+		iframeWin.document.head.appendChild( s );
+	}
+
+	/**
+	 * Ensure stylesheet is present in iframe.
+	 *
+	 * @param {Document} iframeDoc
+	 */
+	function ensureIframeStyles( iframeDoc ) {
+		const parentCss = document.querySelector( 'link[href*="editoria11y"][href$=".css"], link[href*="editoria11y.min.css"]' );
+		if ( parentCss && ! iframeDoc.querySelector( 'link[href="' + parentCss.href + '"]' ) ) {
+			const l = iframeDoc.createElement( 'link' );
+			l.rel = 'stylesheet';
+			l.href = parentCss.href;
+			iframeDoc.head.appendChild( l );
+		}
+	}
+
+	/**
+	 * Create Ed11y instance for an editor iframe.
+	 *
+	 * @param {object} editor
+	 */
+	function initEd11yInstance( editor ) {
+		const iframe = editor.iframeElement || document.getElementById( editor.id + '_ifr' );
+		if ( ! iframe || ! iframe.contentWindow || ! iframe.contentDocument ) {
+			return;
+		}
+		const w = iframe.contentWindow;
+		const d = w.document;
+		if ( ! d.body ) {
+			return setTimeout( function () {
+				initEd11yInstance( editor );
+			}, POLL_INTERVAL );
+		}
+		if ( ! /\btinymce\b/.test( d.body.id ) ) {
+			d.body.id = 'tinymce';
+		}
+		ensureIframeStyles( d );
+		ensureEd11yInIframe( w, function () {
+			if ( ! w.Ed11y ) {
+				return;
+			}
+			if ( w.__pbEd11yInstance ) {
+				return;
+			}
+			const opts = buildEd11yOptions();
+			try {
+				w.__pbEd11yInstance = new w.Ed11y( opts );
+				w.__pbEd11yLastFull = Date.now();
+				w.__pbEd11yVisible = true;
+			} catch ( e ) { /* ignore instantiation error */ }
+		} );
+	}
+
+	/**
+	 * Run full or incremental recheck.
+	 *
+	 * @param {object} editor
+	 * @param {boolean} forceFull
+	 */
+	function runEd11yRecheck( editor, forceFull ) {
+		const iframe = editor.iframeElement || document.getElementById( editor.id + '_ifr' );
+		if ( ! iframe || ! iframe.contentWindow ) {
+			return;
+		}
+		const w = iframe.contentWindow; const inst = w.__pbEd11yInstance; const C = w.Ed11y;
+		if ( ! inst || ! C ) {
+			return;
+		}
+		try {
+			if ( forceFull || Date.now() - ( w.__pbEd11yLastFull || 0 ) > FULL_CHECK_INTERVAL ) {
+				if ( ! C.running && typeof C.checkAll === 'function' ) {
+					C.checkAll(); w.__pbEd11yLastFull = Date.now();
+				}
+			} else if ( ! C.running && typeof C.incrementalCheck === 'function' ) {
+				C.incrementalCheck();
+			} else if ( ! C.running && typeof C.checkAll === 'function' ) {
+				C.checkAll(); w.__pbEd11yLastFull = Date.now();
+			}
+		} catch ( e ) {
+			/* ignore recheck error */
+		}
+	}
+
+	/**
+	 * Hook lifecycle for a TinyMCE editor.
+	 *
+	 * @param {object} editor
+	 */
+	function attachEditorLifecycle( editor ) {
+		if ( initializedEditors.has( editor ) ) {
```

Comment:
Use a WeakSet of initialized editors plus a MutationObserver to reapply hooks only when needed.